index.js

```diff
@@ -99,7 +92,61 @@ async function pay (plugin, {
   }
 }
 
+async function pull (plugin, {
+  pointer,
+  amount,
+  streamOpts = {},
+  callback,
+  callbackOpts = {}
+}) {
+  await plugin.connect()
+  const receiveMax = amount || Infinity
+
+  let response
+  try {
+    response = await query(pointer)
+  } catch (err) {
```

Comment:
I don't understand what this block does

Reply:
It is the logic for pulling from an offline generated pull pointer.

Reply:
so the idea is it's posting to create the pointer and then pulling from it? That seems unnecessary; the pull server could just create the pointer if it sees that it's valid but does not yet exist.

Reply:
Sorry for being so brief before. The idea is that you try to query the pull pointer but in the offline case, that does not exist yet. Thus, you POST to create the pointer and then you query it again to get `destination_account` and `shared_secret`. Are you saying that instead of POSTing, the query command with an unknown token should just try to validate and create it?